### PR TITLE
Blast, aura, sleep_bubble, fire_flower, bombette_breaking effects

### PR DESCRIPTION
**Pull request by JaThePlayer**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 9 functions (+0.08%) on `us`.

File: src/effects/aura.c [hunk 1]
```diff
@@ -1,11 +1,194 @@
...
+Gfx* D_E0076E90[] = { D_090023E0, D_090023E0, D_09002440, D_09002440 };
+Gfx* D_E0076EA0[] = { D_090023F8, D_090023F8, D_09002458, D_09002458 };
+Gfx* D_E0076EB0[] = { D_090023B0, D_090023B0, D_09002410, D_09002410 };
+Gfx* D_E0076EC0[] = { D_090023C8, D_090023C8, D_09002428, D_09002428 };
+Gfx* D_E0076ED0[] = { D_09002000, D_09002000, D_090020E8, D_090021D0 };
+
 void aura_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/aura", aura_main);
+typedef struct AuraFXData {
```
Comment (ethteck) [inline src/effects/aura.c:24]: Effect data structs should go into effects.h (in order, which is kinda annoying atm, sorry for the inconvenience)

Comment (ethteck) [reply src/effects/aura.c:24]: also relevant for your other structs below

File: src/effects/aura.c [hunk 2]
```diff
@@ -1,11 +1,194 @@
...
+    effect = shim_create_effect_instance(bpPtr);
+    effect->numParts = numParts;
+    
+    part = effect->data = shim_general_heap_malloc(numParts * sizeof(*part));
+
+    ASSERT(effect->data != NULL);
+
+    part->unk_64.s = 0;
+    zero = part->unk_64.f;
+    part->unk_40 = zero;
```
Comment (ethteck) [inline src/effects/aura.c:89]: I'm thinking you can get away with using 0 as a literal by doing a chained assignment (thing = other = stuff = 0)..worth a shot

File: src/effects/bombette_breaking.c [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+extern Gfx D_09001D00[];
+extern Gfx D_09001CC8[];
+extern Gfx D_09001C80[];
+extern Gfx D_090017D0[];
+
+Gfx* D_E0084E10[] = { D_090018C0, D_090018C0, D_090018C0 };
+Gfx* D_E0084E1C[] = { D_09001D00, D_09001CC8, D_09001C80 };
+Gfx* D_E0084E28[] = { D_090017D0, D_090017D0, D_090017D0 };
+
+void (*D_E0084E34)(u16, f32*, f32*, f32*, f32*, f32*, f32*) = get_model_center_and_size;
```
Comment (ethteck) [inline src/effects/bombette_breaking.c:20]: neat, more shim functions - except specific to one effect

Comment (ethteck) [reply src/effects/bombette_breaking.c:20]: wait actually can't you just call these functions? D_E0084E3C is func_E0084E3C (you can put a comment above the func name saying it's a copy of xyz func)

Comment (ethteck) [reply src/effects/bombette_breaking.c:20]: nm I can't read. shim functions lol

File: src/effects/sleep_bubble.c [hunk 4]
```diff
@@ -2,13 +2,118 @@
...
+
+    temp1 = part->unk_18 + (part->unk_14 - part->unk_18) * 0.1f;
+    part->unk_18 = temp1;
+    temp2 = temp1 + (temp1 * shim_sin_deg(unk_20 * 3)) * 0.1;
+
+    i = 0;
+    yPtr = &part->points->y;
+    xAngle = yAngle = i;
+
+    do {
```
Comment (ethteck) [inline src/effects/sleep_bubble.c:93]: This is kinda weird. do you have to initialize xPtr way at the beginning but yPtr down right before the loop? I feel like it should be able to be part->points->x[i] and part->points-y[i]

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/effects/aura.c (hunk 2 above)
Comment (JaThePlayer) [reply src/effects/aura.c:89]: It's probably not going to work
image

File: src/effects/sleep_bubble.c (hunk 4 above)
Comment (JaThePlayer) [reply src/effects/sleep_bubble.c:93]: xPtr definitely needs to be initialized at the beginning, I got pretty close to eliminating yPtr (https://decomp.me/scratch/LPgGi) but I'm not sure if there's a way to convince GCC to load part->0x28 sooner

**Review by nanaian — APPROVED** [review]

**Review by ethteck — APPROVED** [review]

**Pull request merged**